mesonbuild/backend/backends.py

```diff
@@ -734,7 +734,7 @@ def create_test_serialisation(self, tests):
                 # E.g. an external verifier or simulator program run on a generated executable.
                 # Can always be run without a wrapper.
                 test_for_machine = MachineChoice.BUILD
-            is_cross = not self.environment.machines.matches_build_machine(test_for_machine)
+            is_cross = self.environment.is_cross_build() if test_for_machine == MachineChoice.HOST else False
```

Comment:
This should only call `is_cross_build`. _All_ logic on deciding whether a build is a cross build _must_ be inside that function. Splitting it around like this is breaking DRY. If you need extra logic then it should go in a different named variable.

Reply:
@jpakkane Great point! Thank you. Just force-pushed a new version.